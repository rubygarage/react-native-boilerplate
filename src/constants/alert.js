// eslint-disable-next-line import/prefer-default-export
export const ALERT_TYPES = {
  error: 'error',
  success: 'success',
  info: 'info',
};
