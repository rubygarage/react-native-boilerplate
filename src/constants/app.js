export const GLOBAL_ERROR = '_error';
export const OFFLINE_FLASH_ID = 'offline';
export const OFFLINE_FLASH_PRIORITY = 0;
export const FLASH_LIFETIME = 3000;

export const PLATFORM_TYPE = {
  android: 'android',
  ios: 'ios',
};
