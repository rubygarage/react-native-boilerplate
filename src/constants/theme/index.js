export default {
  colors: {
    black: '#000000',
    white: '#ffffff',
  },
  fonts: {
  },
  sizes: {
  },
};
