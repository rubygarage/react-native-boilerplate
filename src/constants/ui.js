// eslint-disable-next-line import/prefer-default-export
export const POPOVER_PLACEMENT = {
  bottom: 'bottom',
  auto: 'auto',
};
