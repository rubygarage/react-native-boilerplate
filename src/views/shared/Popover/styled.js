import styled from 'styled-components/native';
import PopoverView from 'react-native-popover-view';

const Popover = styled(PopoverView).attrs(({ theme }) => ({
  arrowStyle: { backgroundColor: 'transparent' },
  backgroundStyle: { backgroundColor: 'transparent' },
  popoverStyle: {
    elevation: 2,
    borderRadius: 4,
    shadowColor: theme.colors.grey900,
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.27,
    shadowRadius: 4.65,
    overflow: 'visible',
  },
}))``;

// eslint-disable-next-line import/prefer-default-export
export { Popover };
