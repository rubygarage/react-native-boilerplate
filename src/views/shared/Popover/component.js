import React from 'react';
import PropTypes from 'prop-types';
import { keys } from 'ramda';

import { POPOVER_PLACEMENT } from 'constants/ui';
import * as S from './styled';

const Popover = ({
  children, isVisible, from, placement, onRequestClose,
}) => (
  <S.Popover
    isVisible={isVisible}
    from={from}
    placement={placement}
    mode="rn-modal"
    onRequestClose={onRequestClose}
  >
    {children}
  </S.Popover>
);

Popover.propTypes = {
  children: PropTypes.node.isRequired,
  isVisible: PropTypes.bool.isRequired,
  placement: PropTypes.oneOf(keys(POPOVER_PLACEMENT)).isRequired,
  onRequestClose: PropTypes.func.isRequired,
  from: PropTypes.shape(),
};

Popover.defaultProps = {
  from: undefined,
};

export default Popover;
