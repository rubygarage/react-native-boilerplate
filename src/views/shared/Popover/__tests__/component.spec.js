import React from 'react';
import { shallow } from 'enzyme';
import { View } from 'react-native';

import { POPOVER_PLACEMENT } from 'constants/ui';
import theme from 'utils/testHelpers/mockedTheme';
import Popover from '../component';

describe('Popover component', () => {
  const defaultProps = {
    isVisible: true,
    fromView: {},
    placement: POPOVER_PLACEMENT.auto,
    onRequestClose: jest.fn(),
    theme,
  };
  const component = shallow(<Popover {...defaultProps}><View /></Popover>);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
