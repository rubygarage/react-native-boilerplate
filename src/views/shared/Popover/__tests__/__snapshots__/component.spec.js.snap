// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component renders correctly 1`] = `
<Styled(Popover)
  isVisible={true}
  mode="rn-modal"
  onRequestClose={[MockFunction]}
  placement="auto"
>
  <View />
</Styled(Popover)>
`;
